html/captive-portal/templates/login.html

```diff
@@ -70,8 +73,15 @@
           <div class="input">
             <span>[% i18n("Password") %]</span><input class="field" name="password" type="password" />
           </div>
+
           [% END %]
-          
+          [% FOREACH field IN mandatory_fields %]
+          <div class="input">
+            <span>[% i18n("$field") %]</span>
+            <input class="field" name="[% field %]" type="text" value="" /><br/>
```

Comment:
value should be the value of the field. This way, if there's an error, the field will not be blank and will be populated by the value entered